samwell/itertools.py

```diff
@@ -132,6 +133,13 @@ def peek(self) -> IterType:
         else:
             raise StopIteration
 
```

Comment:
Can you make a one-liner? Can you add a unit tests testing both sides of the condition?